src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -2361,7 +2361,7 @@ public static string LastWriteTimeString(PSObject instance)
                     // non-existing targets on either Windows or Linux.
```

Comment:
@iSazonov, your last commit had 4 failures in `PowerShell-CI-macos`
Hard link and symbolic link tests.New-Item and hard/symbolic links.New-Item can create a symbolic link to nothing
```powershell
Expected $true, but got $false.
at <ScriptBlock>, /Users/runner/runners/2.169.0/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/FileSystem.Tests.ps1: line 499
499:             Test-Path $symLinkToNothing | Should -BeTrue
```
Hard link and symbolic link tests.New-Item and hard/symbolic links.New-Item emits an error when path to symbolic link already exists.
```powershell
Expected an exception, with FullyQualifiedErrorId 'SymLinkExists,Microsoft.PowerShell.Commands.NewItemCommand' to be thrown, but the FullyQualifiedErrorId was 'AccessException,Microsoft.PowerShell.Commands.NewItemCommand'. from /Users/runner/runners/2.169.0/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/FileSystem.Tests.ps1:505 char:15
    + ?           { New-Item -ItemType SymbolicLink -Path $realDir -Value $sy ?
    +               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
at <ScriptBlock>, /Users/runner/runners/2.169.0/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/FileSystem.Tests.ps1: line 505
505:             { New-Item -ItemType SymbolicLink -Path $realDir -Value $symLinkToDir -ErrorAction Stop } | Should -Throw -ErrorId "SymLinkExists,Microsoft.PowerShell.Commands.NewItemCommand"
```
New-Item with links.Should create a symbolic link to a non-existing file without error
```powershell
Expected $true, but got $false.
at <ScriptBlock>, /Users/runner/runners/2.169.0/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/New-Item.Tests.ps1: line 225
225:         Test-Path $FullyQualifiedLink | Should -BeTrue
```
New-Item with links.New-Item -ItemType SymbolicLink should understand directory path ending with slash
```powershell
Expected a value, but got $null or empty.
at <ScriptBlock>, /Users/runner/runners/2.169.0/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/New-Item.Tests.ps1: line 255
255:         $symbolicLinkPath | Should -Not -BeNullOrEmpty
```